pwiz_tools/Skyline/TestUtil/TestFunctional.cs

```diff
@@ -972,17 +1049,44 @@ private void PauseForScreenShot(string description, int? pageNum, Type formType)
                 Thread.Sleep(Program.PauseSeconds * 1000);
             else if (IsPauseForScreenShots)
             {
+                if (screenshotForm == null)
+                {
+                    screenshotForm = SkylineWindow;
+                    if (formType != null)
+                        screenshotForm = TryWaitForOpenForm(formType);
+                    else if (SkylineWindow.DockPanel.FloatingWindows.Count > 0)
+                    {
+                        screenshotForm = SkylineWindow.DockPanel.FloatingWindows[0];
+                    }
+                    RunUI(() => screenshotForm.Update());
+                }
+
+                Thread.Sleep(300);
+                _shotManager.TakeNextShot(screenshotForm);
+
                 var formSeen = new FormSeen();
                 formSeen.Saw(formType);
-                bool showMathingPages = IsShowMatchingTutorialPages || Program.ShowMatchingPages;
-                PauseAndContinueForm.Show(description, LinkPage(pageNum), showMathingPages);
+                bool showMatchingPages = IsShowMatchingTutorialPages || Program.ShowMatchingPages;
+
+                PauseAndContinueForm.Show(description, LinkPage(pageNum), showMatchingPages);
             }
             else
             {
                 PauseForForm(formType);
             }
         }
 
+        public Form FindFloatingWindow<T>() where T:FloatingWindow
+        {
+            foreach (var panel in SkylineWindow.DockPanel.FloatingWindows)

```

Comment:
Usually you would write something like this in one line:
return SkylineWindow.DockPanel.FloatingWindows.OfType<T>().FirstOrDefault();

Reply:
This is nice, thanks. I didn't know about OfType<>() method.